egret/model_library/unit_commitment/params.py

```diff
@@ -719,8 +719,8 @@ def validate_startup_lags_rule(m, g):
     
     # while startup costs must be monotonically non-decreasing!
     def validate_startup_costs_rule(m, g):
-       startup_costs = list(m.StartupCosts[g])
-       for i in range(0, len(startup_costs)-2):
+       startup_costs = m.StartupCosts[g]
+       for i in range(1, len(startup_costs)-1):
```

Comment:
For some reason, where ever they were used, model.StartupCosts / model.StartupLags were converted to lists. The pyomo sets StartupCosts/StartupLags are ordered, so they're 1-indexed. Obviously lists are 0-indexed, so to avoid conversion in lots of places, this PR uses the original ordered set.

Reply:
Thanks - I didn't recall that Pyomo ordered Sets() were indexed 1-based.